src/fastfield/reader.rs

```diff
@@ -238,7 +290,7 @@ impl<Item: FastValue, C: FastFieldCodecReader + Clone> FastFieldReader<Item>
     /// deleted document, and should be considered as an upper bound
     /// of the actual maximum value.
     fn min_value(&self) -> Item {
-        Item::from_u64(self.reader.min_value())
+        Item::from_u64(self.reader.min_value() * self.gcd + self.min_value)
```

Comment:
Hmmm... Interesting.

So you went for 
` gcd * (min_value + encoded_val)`.

At the same cost, `min_value + gcd * encoded_val` was slightly more general.

(it does not matter though)